src/devtran.cpp

```diff
@@ -559,12 +560,13 @@ Rcpp::List DEVTRAN(const Rcpp::List parin,
             newev->lagged();
             newev->time(this_rec->time() + prob.alag(this_cmtn));
             newev->ss(0);
-            reclist::iterator alagit = a[i].begin()+j;
-            advance(alagit,1);
-            a[i].insert(alagit,newev);
+            //reclist::iterator alagit = a[i].begin()+j;
+            //advance(alagit,1);
+            //a[i].insert(alagit,newev);
+            insert_record(a[i], j, newev, put_ev_first);
             newev->schedule(a[i], maxtime, put_ev_first, NN, prob.alag(this_cmtn));
             this_rec->unarm();
-            sort_recs = true;
+            sort_recs = newev->needs_sorting();
```

Comment:
Needs sorting only if there were additional doses given in `schedule()`. In this case, the records are pushed and we have to sort. I thought this wasn't a huge problem because we usually put in a bunch of doses at once and only have to do it limited number of times, rather every dose. 

Might look at using insert dose here too eventually.